src/Bridge/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaCollectionRestriction.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Validator\Metadata\Property\Restriction;
+
+use ApiPlatform\Core\Metadata\Property\PropertyMetadata;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Constraints\Collection;
+use Symfony\Component\Validator\Constraints\Optional;
+use Symfony\Component\Validator\Constraints\Required;
+use Symfony\Component\Validator\Constraints\Type;
+
+/**
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
+ */
+final class PropertySchemaCollectionRestriction implements PropertySchemaRestrictionMetadataInterface
+{
+    /**
+     * @var iterable<PropertySchemaRestrictionMetadataInterface>
+     */
+    private $restrictionsMetadata;
+
+    /**
+     * @param iterable<PropertySchemaRestrictionMetadataInterface> $restrictionsMetadata
+     */
+    public function __construct(iterable $restrictionsMetadata = [])
+    {
+        $this->restrictionsMetadata = $restrictionsMetadata;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param Collection $constraint
+     */
+    public function create(Constraint $constraint, PropertyMetadata $propertyMetadata): array
+    {
+        $restriction = [
+            'type' => 'object',
+            'properties' => [],
+            'additionalProperties' => $constraint->allowExtraFields,
+        ];
+        $required = [];
+
+        foreach ($constraint->fields as $field => $baseConstraint) {
+            /** @var Required|Optional $baseConstraint */
+            if ($baseConstraint instanceof Required) {
+                $required[] = $field;
+            }
+
+            $restriction['properties'][$field] = $this->mergeConstraintRestrictions($baseConstraint, $propertyMetadata);
+        }
+
+        if ($required) {
+            $restriction['required'] = $required;
+        }
+
+        return $restriction;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Constraint $constraint, PropertyMetadata $propertyMetadata): bool
+    {
+        return $constraint instanceof Collection;
+    }
+
+    /**
+     * @param Required|Optional $constraint
+     */
+    private function mergeConstraintRestrictions(Constraint $constraint, PropertyMetadata $propertyMetadata): array
+    {
+        $propertyRestrictions = [['type' => 'string']];
+
+        foreach ($constraint->getNestedContraints() as $nestedContraint) {
```

Comment:
This causes an error with Symfony 4.4 because `Composite::getNestedContraints()` was only added in 5.1 (but I don't know how to work around, sorry 😖)

Reply:
Yes, I saw the errors, but looking at the source https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Validator/Constraints/Composite.php the method is there :thinking:

Reply:
Another option probably would be directly use `$constraints` property

Reply:
Correction indeed: it was added in v3.4.44, v4.4.12, v5.1.4 and v5.2.0 (the "lowest" test uses symfony/validator v4.4.0)

Reply:
now green :)